pkg/proxy/endpointslicecache.go

```diff
@@ -122,16 +126,21 @@ func newEndpointSliceInfo(endpointSlice *discovery.EndpointSlice, remove bool) *
 
 	if !remove {
 		for _, endpoint := range endpointSlice.Endpoints {
-			if endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready {
-				eInfo := endpointInfo{
-					Addresses: endpoint.Addresses,
-					Topology:  endpoint.Topology,
-				}
-				if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceNodeName) {
-					eInfo.NodeName = endpoint.NodeName
-				}
-				esInfo.Endpoints = append(esInfo.Endpoints, &eInfo)
+			epInfo := &endpointInfo{
+				Addresses: endpoint.Addresses,
+				Topology:  endpoint.Topology,
+
+				// conditions
+				Ready:       endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready,
+				Serving:     endpoint.Conditions.Serving == nil || *endpoint.Conditions.Serving,
+				Terminating: endpoint.Conditions.Terminating != nil && *endpoint.Conditions.Terminating,
```

Comment:
Does this need a feature gate check? (Is it possible you're reading an out-of-date value here if the feature gate was previously on but then turned off?)

Reply:
IMO the cache updates don't need a feature gate since we handle the `nil` case, but the proxier behavior we will end later probably needs one.